ci/build.sh

```diff
@@ -27,20 +27,23 @@ OLD_CWD=$(readlink -f .)
 pushd "$BUILD_DIR"
 
 if [ "$ARCH" == "x86_64" ]; then
+    GET_CMAKE=TRUE
     EXTRA_CMAKE_ARGS=()
 elif [ "$ARCH" == "i386" ]; then
+    GET_CMAKE=TRUE
     EXTRA_CMAKE_ARGS=("-DCMAKE_TOOLCHAIN_FILE=$REPO_ROOT/cmake/toolchains/i386-linux-gnu.cmake" "-DUSE_SYSTEM_CIMG=OFF")
 else
-    echo "Architecture not supported: $ARCH" 1>&2
-    exit 1
+    GET_CMAKE=FALSE
+    EXTRA_CMAKE_ARGS=()
 fi
 
+if [ "$GET_CMAKE" == 'TRUE' ]; then
 # fetch up-to-date CMake
-mkdir cmake-prefix
-wget -O- https://github.com/Kitware/CMake/releases/download/v3.18.1/cmake-3.18.1-Linux-x86_64.tar.gz | tar -xz -C cmake-prefix --strip-components=1
-export PATH="$(readlink -f cmake-prefix/bin):$PATH"
-cmake --version
-
+  mkdir cmake-prefix
+  wget -O- https://github.com/Kitware/CMake/releases/download/v3.18.1/cmake-3.18.1-Linux-x86_64.tar.gz | tar -xz -C cmake-prefix --strip-components=1
```

Comment:
This is a bit of a hack. Just use an up-to-date distro and the system CMake. Since we perform static builds nowadays, one doesn't have to use an outdated CMake.